src/clr-angular/popover/tooltip/providers/tooltip-id.service.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Subject } from 'rxjs';
+
+@Injectable()
+export class TooltipIdService {
```

Comment:
Do we really need a service for this tooltip id?

Reply:
Yes, we need to connect triggers (aria-describedby) with the id of tooltip content.

Reply:
Sorry. I meant to ask why we need an observable for the tooltip id in this service? Can't we have `@Inject(UNIQUE_ID) public tooltipId: string` in the `clr-tooltip` component? So that both tooltip trigger and content can access the same uniq id?

Reply:
If the consumer is using `*clrIfOpen` the content `id` will change when the content is destroyed and re-created.

Reply:
Yes, but that's true because we are injecting the unique id provider in the tooltip-content. If we inject the unique ide provider in the tooltip component itself which contains both trigger and content, there's no need to keep creating new ids, no?

Reply:
Correct. I followed the same pattern that I used in signposts. I may have had a reason for doing it that way in there but don't recall offhand.

If you feel this should change, I'm happy to re-work it and update the testing.

Reply:
I see. I think that could be better because we could get rid of this service. That means we don't have to subscribe to id observable, and the code would be a bit simpler.

Reply:
We still have to inject the id into the child components though. 
Wouldn't that add similar code to inject the parent, get the id and set it in the trigger/content?

I like having the id in a service b/c it doesn't seem like ID belongs on the ClrTooltip. The ClrTooltip doesn't use it in the templates or in its code.

Reply:
You only need to provide it in the tooltip component and only inject it into the tooltip content component and trigger directive. The id doesn't have to change every time content pane opens. As such, we don't need observable there. This could be just simply a string id provided via UNIQUE_ID_PROVIDER. So we don't really need another service to maintain and test.

Reply:
I moved it to `tooltip.ts`